cli/cmd/install.go

```diff
@@ -399,18 +399,23 @@ func renderCRDs(ctx context.Context, k *k8s.KubernetesAPI, w io.Writer, options
 		return err
 	}
 
+	finalValues := charts.MergeMaps(defaultValues, valuesOverrides)
+
 	// If any of the Gateway API CRDs are installed, we default to rendering the
 	// Gateway API CRDs.
 	if k != nil {
 		installed, err := checkGatewayAPICRDs(ctx, k)
 		if err != nil {
 			return err
 		}
-		if installed == Absent {
+		if installed == Absent && finalValues["installGatewayAPI"] == false {
```

Comment:
I think this still will cause an unwanted error if:

* installed == Absent
* installGatewayAPI == false
* enableHttpRoutes == true

I think the correct validation logic would be something like:

```
httpRoutes := finalValues["installGatewayAPI"]
if enableHttpRoutes, ok := finalValues["enableHttpRoutes"]; ok {
  httpRoutes = enableHttpRoutes
}
if installed == Absent {
  if !httpRoutes {
    error("no one is providing Gateway API")
  }
} else if installed == Linkerd {
  if !httpRoutes {
    error("Linkerd was providing Gateway API but this would remove it")
  }
} else if installed == External {
  if httpRoutes {
    error("Gateway API was already present but this would cause Linkerd to overwrite it")
  }
}
```

Reply:
what about the rest of the flags such as `enableTlsRoutes`?